### PR TITLE
Add tests for valid and invalid status list vcs

### PR DESCRIPTION
**Pull request by JSAssassin**

### CODE REVIEW

File: tests/20-main.spec.js [hunk 1]
```diff
@@ -5,26 +5,38 @@ import {
...
-
+documents.set(SUITE_CONTEXT_URL, SUITE_CONTEXT);
 documents.set(SLC.id, SLC);
 
+const didKeyDriver = didKey.driver();
+
 const documentLoader = extendContextLoader(async url => {
-  const doc = documents.get(url);
+  let doc;
+  if(url.startsWith('did:key')) {
```
Comment (dlongley) [inline tests/20-main.spec.js:35]:
```suggestion
  if(url.startsWith('did:key:')) {
```

File: tests/20-main.spec.js [hunk 2]
```diff
@@ -258,6 +270,111 @@ describe('statusTypeMatches', () => {
...
+      verifyStatusListCredential: true
+    });
+    result.verified.should.equal(false);
+    should.exist(result.error);
+    result.error.cause.errors[0].message.should.equal(
+      'No matching proofs found in the given document.');
+  });
+
+  it('should verify an invalid status list vc when ' +
+  '"verifyStatusListCredential" is set to "false"', async () => {
```
Comment (dlongley) [inline tests/20-main.spec.js:344]:
```suggestion
    '"verifyStatusListCredential" is set to "false"', async () => {
```

File: tests/20-main.spec.js [hunk 3]
```diff
@@ -258,6 +270,111 @@ describe('statusTypeMatches', () => {
...
+      documentLoader,
+      verifyStatusListCredential: true
+    });
+    result.verified.should.equal(false);
+    should.exist(result.error);
+    result.error.cause.errors[0].message.should.equal(
+      'No matching proofs found in the given document.');
+  });
+
+  it('should verify an invalid status list vc when ' +
```
Comment (dlongley) [inline tests/20-main.spec.js:343]:
```suggestion
  it('should verify with an invalid status list vc when ' +
```

File: tests/20-main.spec.js [hunk 4]
```diff
@@ -258,6 +270,111 @@ describe('statusTypeMatches', () => {
...
+    const suite = new Ed25519Signature2020();
+    const result = await checkStatus({
+      credential,
+      suite,
+      documentLoader,
+      verifyStatusListCredential: true
+    });
+    should.not.exist(result.error);
+    result.verified.should.equal(true);
+  });
```
Comment (dlongley) [inline tests/20-main.spec.js:303]: We should duplicate the above test without setting `verifyStatusListCredential` at all -- to cover the default case.

File: tests/20-main.spec.js [hunk 5]
```diff
@@ -258,6 +270,111 @@ describe('statusTypeMatches', () => {
...
+
+  it('should fail to verify an invalid status list vc', async () => {
```
Comment (dlongley) [inline tests/20-main.spec.js:305]: We should add another test where the proof isn't deleted, but the SLC is tampered with (changed in some way).

**Review by dlongley — CHANGES_REQUESTED** [review]
Looking good! A few comments to address.

File: tests/20-main.spec.js (hunk 1 above)
Comment (JSAssassin) [reply tests/20-main.spec.js]: Updated [55f51d0](https://github.com/digitalbazaar/vc-status-list/pull/10/commits/55f51d09f840a044257f66443fdb0adf1240090c)

File: tests/20-main.spec.js (hunk 2 above)
Comment (JSAssassin) [reply tests/20-main.spec.js]: Fixed [67b2454](https://github.com/digitalbazaar/vc-status-list/pull/10/commits/67b24548adaff5b0635199613908a9f4def2e62b)

File: tests/20-main.spec.js (hunk 3 above)
Comment (JSAssassin) [reply tests/20-main.spec.js]: Updated

File: tests/20-main.spec.js (hunk 4 above)
Comment (JSAssassin) [reply tests/20-main.spec.js:303]: added [ac847cb](https://github.com/digitalbazaar/vc-status-list/pull/10/commits/ac847cb0d9fdb54d6214ba64f14fc0c083615406)

File: tests/20-main.spec.js (hunk 5 above)
Comment (JSAssassin) [reply tests/20-main.spec.js:337]: Added [292ca49](https://github.com/digitalbazaar/vc-status-list/pull/10/commits/292ca49af0f00b0879d3c990b2778f4a19f5124f)

**Review by dlongley — APPROVED** [review]
Thanks!

**Pull request merged**